eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubmittedVariantDeclusterProcessor.java

```diff
@@ -0,0 +1,114 @@
+            SubmittedVariant submittedVariant = new SubmittedVariant(submittedVariantEntity);
+            List<String> reasons = new ArrayList<>();
+            if (variantNeedsDecluster(dbsnpVariantsWrapper, submittedVariant, reasons)) {
+                decluster(submittedVariantEntity, submittedVariant, operations, submittedVariants, reasons);
+            } else {
+                submittedVariants.add(submittedVariantEntity);
+            }
+        }
+        dbsnpVariantsWrapper.setSubmittedVariants(submittedVariants);
+        dbsnpVariantsWrapper.setOperations(operations);
+        return dbsnpVariantsWrapper;
+    }
+
+    private boolean variantNeedsDecluster(DbsnpVariantsWrapper wrapper, SubmittedVariant submittedVariant, List<String> reasons) {
+        if (!submittedVariant.isAllelesMatch()) {
+            reasons.add(DECLUSTERED_ALLELES_MISMATCH);
+        }
+        boolean typeMatches = isSameType(wrapper.getClusteredVariant(), submittedVariant,
+                                         wrapper.getDbsnpVariantType());
+        if (!typeMatches) {
+            reasons.add(DECLUSTERED_TYPE_MISMATCH);
+        }
+        return !reasons.isEmpty();
+    }
+
+    private boolean isSameType(DbsnpClusteredVariantEntity clusteredVariant, SubmittedVariant submittedVariant,
+                               DbsnpVariantType dbsnpVariantType) {
+        return VariantClassifier.getVariantClassification(submittedVariant.getReferenceAllele(),
+                                                          submittedVariant.getAlternateAllele(),
+                                                          dbsnpVariantType.intValue())
+                                .equals(clusteredVariant.getType());
+    }
+
+    private void decluster(DbsnpSubmittedVariantEntity nonDeclusteredVariantEntity, SubmittedVariant variant,
+                           List<DbsnpSubmittedVariantOperationEntity> operations,
+                           List<DbsnpSubmittedVariantEntity> submittedVariants,
+                           List<String> reasons) {
+        //Register submitted variant decluster operation
+        DbsnpSubmittedVariantOperationEntity operation = new DbsnpSubmittedVariantOperationEntity();
+        DbsnpSubmittedVariantInactiveEntity inactiveEntity =
+                new DbsnpSubmittedVariantInactiveEntity(nonDeclusteredVariantEntity);
+
+        StringBuilder reason = new StringBuilder(DECLUSTERED);
+        reasons.forEach(reason::append);
+        Long accession = nonDeclusteredVariantEntity.getAccession();
+        operation.fill(EventType.UPDATED, accession, null, reason.toString(),
+                       Collections.singletonList(inactiveEntity));
+        operations.add(operation);
+
+        //Decluster submitted variant
+        variant.setClusteredVariantAccession(null);
+        String hash = nonDeclusteredVariantEntity.getHashedMessage();
```

Comment:
Agreed offline to recalculate hash just in case the summary function is modified in the future to include the RS accession.